### PR TITLE
Docs for request logging

### PR DESCRIPTION
**Pull request by vtlim**
This PR:
* Adds details on request logging including an overview guide and example.
* Cleans up some of the associated reference doc to use more consistent terminology and style

This PR has:
- [x] been self-reviewed.

### CODE REVIEW

File: docs/operations/request-logging.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+All Apache Druid services that can serve queries can also log the query requests they see.
+Request logs contain information on query metrics, including execution time and memory usage.
+You can use information in the request logs to monitor query performance, determine bottlenecks, and analyze and improve slow queries.
+
+Request logging is disabled by default.
+This topic describes how to configure Druid to log and emit query metrics.
```
Comment (TSFenwick) [inline docs/operations/request-logging.md:31]: Is this query metrics? Its more of query logs with associated metrics and dimensions
@suneet-s

Comment (vtlim) [reply docs/operations/request-logging.md:31]: The logs contain the metrics like [`sqlQuery/time`](https://druid.apache.org/docs/latest/operations/metrics.html#sql-metrics) so not sure if this is a matter of semantics. 🤷 
If it's clearer, we could change the phrasing to be something like the following:
```suggestion
This topic describes how to configure Druid to generate request logs to track query metrics.
```

Comment (TSFenwick) [reply docs/operations/request-logging.md:31]: thats better

**Review by TSFenwick — APPROVED** [review]

File: docs/configuration/index.md [hunk 2]
```diff
@@ -279,24 +280,24 @@ For SQL query request, the `native_query` field is empty. Example
 2019-01-14T10:00:00.000Z        127.0.0.1       {"sqlQuery/time":100,"sqlQuery/bytes":600,"success":true,"identity":"user1"}  {"query":"SELECT page, COUNT(*) AS Edits FROM wikiticker WHERE __time BETWEEN TIMESTAMP '2015-09-12 00:00:00' AND TIMESTAMP '2015-09-13 00:00:00' GROUP BY page ORDER BY Edits DESC LIMIT 10","context":{"sqlQueryId":"c9d035a0-5ffd-4a79-a865-3ffdadbb5fdd","nativeQueryIds":"[490978e4-f5c7-4cf6-b174-346e63cf8863]"}}
 ```
 
-#### Emitter Request Logging
+#### Emitter request logging
 
-Every request is emitted to some external location.
+The `emitter` request logger emits every request to some external location, which is configured through an [emitter](#enabling-metrics).
```
Comment (techdocsmith) [inline docs/configuration/index.md:285]:
```suggestion
The `emitter` request logger emits every request to the external location specified in the [emitter](#enabling-metrics) configuration.
```

File: docs/configuration/index.md [hunk 3]
```diff
@@ -279,24 +280,24 @@ For SQL query request, the `native_query` field is empty. Example
...
 
 |Property|Description|Default|
 |--------|-----------|-------|
 |`druid.request.logging.feed`|Feed name for requests.|none|
 
-#### SLF4J Request Logging
+#### SLF4J request logging
 
-Every request is logged via SLF4J. Native queries are serialized into JSON in the log message regardless of the SLF4J format specification. They will be logged under the class `org.apache.druid.server.log.LoggingRequestLogger`.
+The `slf4j` request logger logs every request via  SLF4J. Native queries are serialized into JSON in the log message regardless of the SLF4J format specification. Requests are logged under the class `org.apache.druid.server.log.LoggingRequestLogger`.
```
Comment (techdocsmith) [inline docs/configuration/index.md:293]:
```suggestion
The `slf4j` request logger logs every request using SLF4J. It serializes native queries into JSON in the log message regardless of the SLF4J format specification. Requests are logged under the class `org.apache.druid.server.log.LoggingRequestLogger`.
```

File: docs/configuration/index.md [hunk 4]
```diff
@@ -279,24 +280,24 @@ For SQL query request, the `native_query` field is empty. Example
...
 
 |Property|Description|Default|
 |--------|-----------|-------|
-|`druid.request.logging.setMDC`|If MDC entries should be set in the log entry. Your logging setup still has to be configured to handle MDC to format this data|false|
-|`druid.request.logging.setContextMDC`|If the druid query `context` should be added to the MDC entries. Has no effect unless `setMDC` is `true`|false|
+|`druid.request.logging.setMDC`|If MDC entries should be set in the log entry. Your logging setup still has to be configured to handle MDC to format this data.|false|
```
Comment (techdocsmith) [inline docs/configuration/index.md:297]:
```suggestion
|`druid.request.logging.setMDC`|If you want to set MDC entries within the log entry, set this value to "true"  to handle the MDC data format.|false|
```
not  !00% sure  what this was trying to say, verify before accepting change.

File: docs/configuration/index.md [hunk 5]
```diff
@@ -279,24 +280,24 @@ For SQL query request, the `native_query` field is empty. Example
...
+|`druid.request.logging.setContextMDC`|If the Druid query `context` should be added to the MDC entries. Has no effect unless `setMDC` is `true`.|false|
```
Comment (techdocsmith) [inline docs/configuration/index.md:298]:
```suggestion
|`druid.request.logging.setContextMDC`|Set to "true" to add  the Druid query `context` to the MDC entries. Only applies when `setMDC` is `true`.|false|
```

File: docs/configuration/index.md [hunk 6]
```diff
@@ -310,31 +311,36 @@ For native query, the following MDC fields are populated with `setMDC`:
 |`resultOrdering`|The ordering of results|
 |`descending`|If the query is a descending query|
 
-#### Filtered Request Logging
-Filtered Request Logger filters requests based on a configurable query/time threshold (for native query) and sqlQuery/time threshold (for SQL query).
-For native query, only request logs where query/time is above the threshold are emitted. For SQL query, only request logs where sqlQuery/time is above the threshold are emitted.
+#### Filtered request logging
+
+The `filtered` request logger filters requests based on the query type or how long a query takes to complete.
+For native queries, requests are only logged when the `query/time` metric is above the user-provided threshold.
```
Comment (techdocsmith) [inline docs/configuration/index.md:317]:
```suggestion
For native queries, the logger only logs requests when the `query/time` metric exceeds the user-provided threshold.
```

File: docs/configuration/index.md [hunk 7]
```diff
@@ -310,31 +311,36 @@ For native query, the following MDC fields are populated with `setMDC`:
...
+For SQL queries, requests are only logged when the `sqlQuery/time` metric is above the user-provided threshold.
```
Comment (techdocsmith) [inline docs/configuration/index.md:318]:
```suggestion
For SQL queries, it only logs requests when the `sqlQuery/time` metric exceeds the user-provided threshold.
```
for line 317 too,  is this a specific threshold we can mention. For example when sqlQuery/time exceeds the threshold set in <config>

File: docs/configuration/index.md [hunk 8]
```diff
@@ -454,7 +465,7 @@ The additional configs are:
 
 ##### Graphite Emitter
 
-To use graphite as emitter set `druid.emitter=graphite`. For configuration details please follow this [link](../development/extensions-contrib/graphite.md).
+To use graphite as emitter set `druid.emitter=graphite`. For configuration details, see the [documentation](../development/extensions-contrib/graphite.md) for the Graphite emitter Druid extension.
```
Comment (techdocsmith) [inline docs/configuration/index.md:468]:
```suggestion
To use graphite as emitter set `druid.emitter=graphite`. For configuration details, see [Graphite emitter](../development/extensions-contrib/graphite.md) for the Graphite emitter Druid extension.
```

File: docs/operations/metrics.md [hunk 9]
```diff
@@ -72,11 +69,11 @@ Available Metrics
...
 |`query/cpu/time`|Microseconds of CPU time taken to complete a query|Common: dataSource, type, interval, hasFilters, duration, context, remoteAddress, id. Aggregation Queries: numMetrics, numComplexMetrics. GroupBy: numDimensions. TopN: threshold, dimension.|Varies|
-|`query/count`|number of total queries|This metric is only available if the QueryCountStatsMonitor module is included.||
-|`query/success/count`|number of queries successfully processed|This metric is only available if the QueryCountStatsMonitor module is included.||
-|`query/failed/count`|number of failed queries|This metric is only available if the QueryCountStatsMonitor module is included.||
-|`query/interrupted/count`|number of queries interrupted due to cancellation.|This metric is only available if the QueryCountStatsMonitor module is included.||
-|`query/timeout/count`|number of timed out queries.|This metric is only available if the QueryCountStatsMonitor module is included.||
+|`query/count`|Number of total queries|This metric is only available if the QueryCountStatsMonitor module is included.||
```
Comment (techdocsmith) [inline docs/operations/metrics.md:72]: Number of total reads awkwardly is this the "total number of queries"?

File: docs/operations/request-logging.md [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+All Apache Druid services that can serve queries can also log the query requests they see.
```
Comment (techdocsmith) [inline docs/operations/request-logging.md:26]:
```suggestion
All Apache Druid services that can serve queries can also log the query requests they process.
```
Not sure how sight figures in.

File: docs/operations/request-logging.md [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+See the [Request logging configuration](../configuration/index.md#request-logging) for properties to set for each type of request logger.
+Specify these properties in the `common.runtime.properties` file.
+You must restart Druid for the changes to take effect.
+
+Druid stores the results in the Broker logs, unless the request logging type is `emitter`.
+If you use emitter request logging, you must also configure metrics emission.
+
+## Configure metrics emission
+
+Druid emits metrics and alerts via an emitter.
```
Comment (techdocsmith) [inline docs/operations/request-logging.md:57]:
```suggestion
Druid includes various emitters to send metrics and alerts. 
```

File: docs/operations/request-logging.md [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+
+# Enable metrics emission and tell Druid where to emit messages
+druid.emitter=http
+druid.emitter.http.recipientBaseUrl=http://example.com:8080/path
+
+# Authenticate to the base URL, if needed
+druid.emitter.http.basicAuthentication=username:password
+```
+
+The following output shows an example of the emitted information:
```
Comment (techdocsmith) [inline docs/operations/request-logging.md:90]:
```suggestion
The following shows an example log emitter output:
```

**Review by techdocsmith — CHANGES_REQUESTED** [review]
Mostly style changes.

File: docs/configuration/index.md (hunk 6 above)
Comment (vtlim) [reply docs/configuration/index.md:317]:
```suggestion
For native queries, the logger only logs requests when the `query/time` metric exceeds the threshold provided in `queryTimeThresholdMs`.
```

File: docs/configuration/index.md (hunk 7 above)
Comment (vtlim) [reply docs/configuration/index.md:318]:
```suggestion
For SQL queries, it only logs requests when the `sqlQuery/time` metric exceeds threshold provided in `sqlQueryTimeThresholdMs`.
```

File: docs/operations/metrics.md (hunk 9 above)
Comment (vtlim) [reply docs/operations/metrics.md:72]:
```suggestion
|`query/count`|Total number of queries|This metric is only available if the QueryCountStatsMonitor module is included.||
```

File: docs/configuration/index.md (hunk 4 above)
Comment (vtlim) [reply docs/configuration/index.md:297]:
```suggestion
|`druid.request.logging.setMDC`|If you want to set MDC entries within the log entry, set this value to `true`. Your logging system must be configured to support MDC to format this data.|false|
```

**Review by techdocsmith — APPROVED** [review]
LGTM

**Pull request merged**